kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java

```diff
@@ -121,12 +122,13 @@ public void start() throws Exception {
     }
 
     if (roles.contains(KaldbConfigs.NodeRole.QUERY)) {
-      KaldbDistributedQueryService searcher = new KaldbDistributedQueryService();
-      KaldbDistributedQueryService.servers =
-          new ArrayList<>(KaldbConfig.get().getQueryConfig().getTmpIndexNodesList());
+      searchMetadataStore = new SearchMetadataStore(metadataStore, true);
```

Comment:
this is kinda ugly.. To all other "services" ( guava ) we can get away by passing `metadataStore` and then within the service we can create RecoveryNodeMetadataStore or SearchMetadataStore or some other store

Reply:
But here since `KaldbDistributedQueryService` is a GRPC service so SearchMetadataStore needs to be created in `Kaldb` so that we can use the shutdown hook to close it